framework/src/org/apache/cordova/CordovaInterfaceImpl.java

```diff
@@ -161,4 +164,35 @@ public ActivityResultHolder(int requestCode, int resultCode, Intent intent) {
             this.intent = intent;
         }
     }
+
+    /**
+     * Called by the system when the user grants permissions
+     *
+     * @param requestCode
+     * @param permissions
+     * @param grantResults
+     */
+    public void onRequestPermissionResult(int requestCode, String[] permissions,
+                                          int[] grantResults) throws JSONException {
+        if(permissionResultCallback != null)
+        {
+            permissionResultCallback.onRequestPermissionResult(requestCode, permissions, grantResults);
+            permissionResultCallback = null;
+        }
+    }
+
+    public void requestPermission(CordovaPlugin plugin, int requestCode, String permission) {
+        permissionResultCallback = plugin;
+        String[] permissions = new String [1];
+        permissions[0] = permission;
+        getActivity().requestPermissions(permissions, requestCode);
+    }
+
+    public void requestPermissions(CordovaPlugin plugin, int requestCode)
+    {
+        permissionResultCallback = plugin;
+        String[] permissions = plugin.getPermissionRequest();
```

Comment:
It's kinda odd to have the permission array not being passed along instead we invoking a callback on plugin to get the permissions. Passing the permissions array would be a simpler design.